src/FileProvider/Compat.cs

```diff
@@ -2,27 +2,48 @@
 
 using System;
 using ObjCRuntime;
+using System.Runtime.Versioning;
 
 namespace FileProvider {
 
 #if MONOMAC
 	public static partial class NSFileProviderItem_Extensions {
 
+#if !NET
 		[Obsolete ("This API was removed from macOS in Xcode 12.2 beta 2.")]
+#else
+		[UnsupportedOSPlatform ("ios")]
+		[UnsupportedOSPlatform ("tvos")]
+		[UnsupportedOSPlatform ("macos")]
+		[UnsupportedOSPlatform ("maccatalyst")]
```

Comment:
The unsupported that are not macOS are not needed because it is inside a #if MONOMAC